Реализовано задание спринта 22

DesidiamS opened the pull request.

Commit:
Реализовано задание спринта 22

bismuthD commented on infra/gateway/pom.xml:
```diff
@@ -0,0 +1,66 @@
+    </dependencies>
+
+
+    <properties>
```
Свойства с версией и кодировкой лучше вынести в самый корневой pom, т.к. они будут одинаковые у всех модулей

bismuthD commented on commerce/interaction-api/src/main/java/ru/yandex/practicum/feign/DeliveryFeign.java:
```diff
@@ -0,0 +1,30 @@
+public interface DeliveryFeign {
+
+    @PutMapping
+    DeliveryDto createDelivery(@RequestBody DeliveryDto deliveryDto);
```
стоит добавить @Valid

bismuthD commented on commerce/interaction-api/src/main/java/ru/yandex/practicum/feign/DeliveryFeign.java:
```diff
@@ -0,0 +1,30 @@
+    void failDelivery(@RequestBody UUID deliveryId);
+
+    @PostMapping("/cost")
+    BigDecimal calculateDelivery(@RequestBody OrderDto orderDto);
```
стоит добавить @Valid
Для остальных интерфейсов-клиентов тоже добавь валидацию

bismuthD commented on commerce/delivery/src/main/java/ru/yandex/practicum/exception/DeliveryErrorHandler.java:
```diff
@@ -0,0 +1,27 @@
+        return new ResponseEntity<>(
+                new ErrorResponse(
+                        e.getCause(),
+                        Arrays.asList(e.getStackTrace()),
```
e.getStackTrace() отдавать наружу не очень хорошая идея с точки зрения безопасности, т.к. в нем могут быть названия таблиц, имена пользователей, куски sql т.е. масса инфы которой могут воспользоваться злоумышленники. А вот добавить логирование в методы хендлера, куда и вывести стектрейс, будет хорошим решением.

bismuthD reviewed (commented):
Отличная работа. Архитектура, реализация модулей payment, delivery и order реализованы корректно. Оставил несколько комментариев для правок

Commit:
- Добавлена валидация в feign клиентах
- Убраны stack trace из выдачи ошибок
- Добавлено логирование в классах handler
- Перенесены версии в корневой pom
